pkg/controller/controlplanecerts/controlplanecerts_controller.go

```diff
@@ -363,7 +363,7 @@ func (r *ReconcileControlPlaneCerts) getServingCertificatesJSONPatch(cd *hivev1.
 			additional.Domain, remoteSecretName(bundle.CertificateSecretRef.Name, cd)))
 	}
 
-	var kubeAPIServerNamedCertsTemplate = `[ { "op": "replace", "path": "/spec/servingCerts/namedCertificates", "value": [ %s ] } ]`
+	var kubeAPIServerNamedCertsTemplate = `[ { "op": "add", "path": "/spec/servingCerts", "value": {} }, { "op": "add", "path": "/spec/servingCerts/namedCertificates", "value": [  ] }, { "op": "replace", "path": "/spec/servingCerts/namedCertificates", "value": [ %s ] } ]`
```

Comment:
nit: why can't we combine the two add operations into one so that we add `{"nameCertificates": []}` to `/spec/servingCerts` ?

Reply:
doing some testing,
```json
{
	"spec": {
		"servingCerts": {
			"another key": "some value"
		}
	}
}
```

Applying this patch will,

```
[{
	"op": "add",
	"path": "/spec/servingCerts",
	"value": {}
}, {
	"op": "add",
	"path": "/spec/servingCerts/namedCertificates",
	"value": []
}, {
	"op": "replace",
	"path": "/spec/servingCerts/namedCertificates",
	"value": [{
		"domain": "test-api.com",
		"secret": "some-secret-test-api-com"
	}]
}]
```

will create new object without the `another key` ??
```
{
  "spec" : {
    "servingCertificates" : {
      "another key" : "some value"
    },
    "servingCerts" : {
      "namedCertificates" : [ {
        "domain" : "test-api.com",
        "secret" : "some-secret-test-api-com"
      } ]
    }
  }
}
```

I tried it out here, https://json-schema-validator.herokuapp.com/jsonpatch.jsp

I don't think that is correct, right?

Reply:
I tried this and got a failure as well, it doesn't like sections not existing and it turns out spec.servingCerts is not set either in this scenario, so it seemed like I needed to add it explicitly first.

Reply:
@abhinavdahiya previously this assumed full control over the entire APIServer resource, and this transitions us to assuming control over just the servingCerts and below. It is true that any user attempts to edit other serving cert fields would get overwritten, but I think this is acceptable.

I'm open to work around it if there's a reasonable way to do so but I don't see how to conditionally add spec.servingCerts if it's missing in a json patch. Any ideas?

Reply:
>  this transitions us to assuming control over just the servingCerts and below. It is true that any user attempts to edit other serving cert fields would get overwritten, but I think this is acceptable

Let's move this to the PR descriptions/ commit message because the current description imo does not make that clear. And then we can move ahead and merge it.